modin/core/dataframe/pandas/exchange/dataframe_protocol/from_dataframe.py

```diff
@@ -367,10 +398,9 @@ def bitmask_to_bool_ndarray(
     -------
     np.ndarray[bool]
     """
-    if first_byte_offset > 8:
-        raise ValueError(
-            f"First byte offset can't be more than 8, met: {first_byte_offset}"
-        )
+    bytes_to_skip = first_byte_offset // 8
+    bitmask = bitmask[bytes_to_skip:]
```

Comment:
if you do so don't you have to adjust `mask_length`, too? also I don't see a need of copying this thing, you can just make a variable storing "index of first byte" and pass it as `start=` to subsequent `range()` call

Reply:
> if you do so don't you have to adjust mask_length, too?

No, `mask_length` is basically a `col.size` here, so it means the number of bool elements (bits) including an offset

> I don't see a need of copying this thing

Numpy doesn't copies the data on simple slices:
```python
>>> arr1 = np.array([1, 2, 3, 4, 5], dtype=np.uint8)
>>> arr2 = arr1[2:]
>>> (arr1.__array_interface__["data"][0] + 2) == arr2.__array_interface__["data"][0]
True
```

> make a variable storing "index of first byte"

using a separate variable for a start byte introduces more magic arithmetics which is much error-prone rather than having already sliced `bitmask`:
```diff

-    val = bitmask[0]
+    val = bitmask[first_byte_idx]
     mask_idx = 0
     bits_in_first_byte = min(8 - first_byte_offset, mask_length)
     for j in range(bits_in_first_byte):
@@ -416,13 +417,13 @@ def bitmask_to_bool_ndarray(
     # `mask_length // 8` describes how many full bytes to process
     for i in range((mask_length - bits_in_first_byte) // 8):
         # doing `+ 1` as we already processed the first byte
-        val = bitmask[i + 1]
+        val = bitmask[first_byte_idx + i + 1]
         for j in range(8):
             if val & (1 << j):
                 bool_mask[mask_idx] = True
             mask_idx += 1

-    if len(bitmask) > 1:
+    if (len(bitmask) - first_byte_idx) > 1:
```